x-pack/packages/security/ui_components/src/kibana_privilege_table/sub_feature_form.tsx

```diff
@@ -33,10 +33,23 @@ interface Props {
   privilegeIndex: number;
   onChange: (selectedPrivileges: string[]) => void;
   disabled?: boolean;
+  categoryId?: string;
 }
 
 export const SubFeatureForm = (props: Props) => {
   const groupsWithPrivileges = props.subFeature.getPrivilegeGroups();
+  const subFeatureNameTestId = useMemo(() => {
+    // Removes anything that is not a Number, Letter or space and replaces it with _
+    return props.subFeature.name.toLowerCase().replace(/[^\w\d]/g, '_');
+  }, [props.subFeature.name]);
+  const getTestId = useCallback(
+    (suffix: string = '') => {
+      return `${props.categoryId ? `${props.categoryId}_` : ''}${
+        props.featureId
+      }_${subFeatureNameTestId}${suffix ? `_${suffix}` : ''}`;
```

Comment:
I always opt against having generators for test-ids because it's almost impossible to find the element from the web console in the codebase. What do you say if we hardcode the test-ids?

Reply:
I'm actually of the opposite opinion. Having static test ids in some cases makes it harder to find the elements you are after, especially when a component is reused multiple times on the page - as is the case here.

I rather keep it as is, unless you have a better way to select elements on the page in a more efficient way

Reply:
Whenever I want to debug functionality, I get the data-test-id from devconsole. If it was hardcoded I could just use Ctrl+F and search for it, with the prefixes generators etc it's almost impossible - and have to look for chunks and later confirm if I found a correct prefix. 

However, I don't insist, just sharing personal opinion :P It can stay this way 👍 Thanks!

Reply:
So yeah, I do think we differ in the way we approach test development. If I'm working on e2e tests, I do in fact use the browser's developer tools to grap the `data-test-subj`'s  for elements I need to access. And that is what these changes are doing to the role forms. And in this case, because those forms are used multiple times on the page, using a static `data-test-subj` would negate the purpose as to why I made them - to facilitate and more predictably select elements in DOM.

I will leave it as is since you are ok with it.

Reply:
Of course, all good 👍 thanks for the consideration :)